pallets/staking/src/lib.rs

```diff
@@ -1543,6 +1559,23 @@ decl_module! {
 
         fn deposit_event() = default;
 
+        fn on_runtime_upgrade() -> Weight {
+            use frame_support::storage::migration::{ StorageIterator, put_storage_value };
+
+            if StorageVersion::get() == Releases::V4_0_0 {
+                StorageIterator::<()>::new(b"Staking", b"PermissionedValidators")
+                    .drain()
+                    .filter_map(|(key, value)| {
+                        T::AccountId::decode(&mut Twox64Concat::reverse(&key)).map_or(None, |id| {
+                            Some((<Identity<T>>::get_identity(&id).unwrap_or_default(), value))
+                        })
+                    })
+                    .for_each(|(key, value)| put_storage_value(b"Staking", b"PermissionedEntities", &key.as_bytes(), value));
```

Comment:
You could either type `key` and `value` explicitly or rename those to something that tells about their types, possibly `did` and `account_id`?